### PR TITLE
Randomize naming for createVolumeDirectory method

### PR DESCRIPTION
**Pull request by johnathana**
Currently createVolumeDirectory method uses `System.currentTimeMillis()` to randomize the volume directory name. This appears to be problematic for test suites running in parallel, falling to the exact same millisecond causing random test failures.

### CODE REVIEW

**Review by rnorth — APPROVED** [review]
Thanks @johnathana, this seems like a sensible fix to apply

**Comment by bsideup** [comment]
@rnorth btw we should mark it as deprecated as the method is unused

**Comment by johnathana** [comment]
> @rnorth btw we should mark it as deprecated as the method is unused

This is a nice utility function. I use it to extract jar classhpath resources in CI docker in docker environments.

**Comment by bsideup** [comment]
@johnathana consider using `MountableFile` abstraction for that :)

**Comment by johnathana** [comment]
> @johnathana consider using `MountableFile` abstraction for that :)

Thank you for your prompt response @bsideup. I am already using `MountableFile` and the `getResolvedPath` method to do all the heavy lifting, extracting to a temp dir. In CI envs and specifically Jenkins, docker in docker can be tricky. So I want files to the CWD. The final code looks like that: 

```
def extractClasspathResourceFolderToCwd(resourceFolder: String): String = {
  val resourcePath      = MountableFile.forClasspathResource(resourceFolder).getResolvedPath
  val destinationFolder = createVolumeDirectory(true)
  FileUtils.copyDirectory(new File(resourcePath), destinationFolder.toFile)
  destinationFolder.toAbsolutePath.toString
}
```

**Comment by bsideup** [comment]
@johnathana since creating temporary files in a current directory isn't Testcontainers' responsibility, I'd still prefer to remote that unused method, especially given how simple it is.

File: core/src/main/java/org/testcontainers/containers/GenericContainer.java [hunk 1]
```diff
@@ -623,7 +624,7 @@ protected Logger logger() {
      * @return path to the volume directory
      */
     protected Path createVolumeDirectory(boolean temporary) {
```
Comment (rnorth) [inline core/src/main/java/org/testcontainers/containers/GenericContainer.java:626]:
```suggestion
    @Deprecated
    protected Path createVolumeDirectory(boolean temporary) {
```

**Review by kiview — APPROVED** [review]
Thanks for the fix and adding the deprecation notice @johnathana.

**Pull request merged**